### ISSUE TITLE
Error: ENOENT: no such file or directory, open ../snapshot-db_name.json

### ISSUE THREAD

**Issue by Juknum** [mikro-orm/mikro-orm#4306]
**Describe the bug**
Hi, I would like to run the `npx mikro-orm migration:create --initial` cli command but I'm stuck on a ENOENT error;
> By the way I would like to set up a "base" for my database when I first init my project, is it the right way to do it ? 

**Stack trace**
```bash
mikro-orm migration:create

Create new migration with current schema diff

Options:
  -b, --blank    Create blank migration                                [boolean]
  -i, --initial  Create initial migration                              [boolean]
  -d, --dump     Dumps all queries to console                          [boolean]
  -p, --path     Sets path to directory where to save entities          [string]
  -v, --version  Show version number                                   [boolean]
  -h, --help     Show help                                             [boolean]

[Error: ENOENT: no such file or directory, open '/Users/juknum/Documents/Code/AE/api/src/migrations/.snapshot-ae_test.json'] {
  errno: -2,
  code: 'ENOENT',
  syscall: 'open',
  path: '/Users/juknum/Documents/Code/AE/api/src/migrations/.snapshot-ae_test.json'
}
```

**To Reproduce**
1. Run `npx mikro-orm migration:create --initial`

**Expected behavior**
The migration class file should be created ?

**Additional context**

I'm using Mikro-ORM with NestJS & GraphQL ([GitHub repo](https://github.com/ae-utbm/api/tree/setup-mikrorm))
/ and I use the following config file :

```ts
// mikro-orm.config.ts
import { MikroORMOptions, IDatabaseDriver, Connection } from '@mikro-orm/core';
import { TsMorphMetadataProvider } from '@mikro-orm/reflection';
import { Logger } from '@nestjs/common';
import { SqlHighlighter } from '@mikro-orm/sql-highlighter';

const logger = new Logger('MikroORM');

export default (): Partial<MikroORMOptions<IDatabaseDriver<Connection>>> => ({
	type: 'postgresql',
	dbName: process.env.DB_NAME, // 'postgres'
	port: parseInt(process.env.DB_PORT, 10), // 5432
	host: process.env.DB_HOST, // '127.0.0.1'
	user: process.env.DB_USER, // 'root'
	debug: process.env.DEBUG === 'true',
	entities: ['./dist/modules/**/entities/*.entity.js'],
	entitiesTs: ['./src/modules/**/entities/*.entity.ts'],
	highlighter: new SqlHighlighter(),
	migrations: {
		snapshot: false,
		transactional: true,
		path: './dist/database/migrations',
		pathTs: './src/database/migrations',
	},
	logger: logger.log.bind(logger),
	metadataProvider: TsMorphMetadataProvider,
});
```

**Versions**

| Dependency | Version |
| - | - |
| node | `20.0.0` |
| typescript | `^4.9.5` |
| mikro-orm | `^5.7.4` |
| your-driver | ? |

**Comment by B4nan** [comment]
Does it work if you drop the `--initial`? You almost never need this, only in case you already have an existing schema.

> snapshot: false

Just curious, did you add this when trying to get around the error? You should generally keep snapshots enabled. It's indeed weird that it tries to load the snapshot even when it's disabled.

**Comment by Juknum** [comment]
Yup I have the `No changes required, schema is up-to-date` message but I don't have any migration file created.

And yeah I have added the snapshot settings as a workaround but nothing changed :/

**Comment by B4nan** [comment]
Can you try to add this line [here](https://github.com/mikro-orm/mikro-orm/blob/master/packages/migrations/src/Migrator.ts#L376)?

```diff
private async ensureMigrationsDirExists() {
  if (!this.options.migrationsList) {
    await ensureDir(this.absolutePath);
+    await ensureDir(this.snapshotPath);
  }
}
```

**Comment by B4nan** [comment]
That's really weird, try to find the place that is causing the error. I checked the code, and all places I see that are trying to read or write the snapshot are both checking the option you have disabled, as well as the file existence itself. Only the `Migrator` class should be working with the snapshot.

**Comment by B4nan** [comment]
Also the fact that it tries to open file `'/Users/juknum/Documents/Code/AE/api/src/migrations/.snapshot-ae_test.json'` is weird - it should respect the migration paths you set to `'./src/database/migrations'`:

https://github.com/mikro-orm/mikro-orm/blob/master/packages/migrations/src/Migrator.ts#L35-L39

**Comment by Juknum** [comment]
Yeah that's weird, I don't think my config does anything, when I add a `console.log(this.options)` I get the following : 
```js
{
  tableName: 'mikro_orm_migrations',
  path: './dist/migrations',
  glob: '!(*.d).{js,ts,cjs}',
  silent: false,
  transactional: true,
  disableForeignKeys: true,
  allOrNothing: true,
  dropTables: true,
  safe: false,
  snapshot: true,
  emit: 'ts',
  fileName: [Function: fileName],
  pathTs: './src/migrations'
}
``` 

Which looks like a default config file, is there anything I am missing ?

**Comment by B4nan** [comment]
Yes, I see you skipped that. I think the `useTsNode` option could be enough, that will enable look up of the TS config. The `src/dist` folders are nowadays checked automatically, but only if you are inside ts-node context. The CLI won't register ts-node without you enabling the `useTsNode` flag (and installing ts-node, which I see you already have).

```json
  "name": "our-app",
  "dependencies": { ... },
  "mikro-orm": {
    "useTsNode": true,
    "configPaths": [
      "./src/mikro-orm.config.ts",
      "./dist/mikro-orm.config.js"
    ]
  }
}
```

https://mikro-orm.io/docs/installation#setting-up-the-commandline-tool

**Comment by B4nan** [comment]
Yeah, this is the PR that handles the smarter dynamic defaults #3497, and it explicitly says the `useTsNode` is still required.

**Comment by B4nan** [comment]
> import { BaseEntity } from 'src/database/entities/base.entity';

This is not an absolute path, this is paths mapping - you are apparently using tsconfig-paths, and you need to configure them. The CLI does support that too, as long as you defined them in the root tsconfig.json (or you set the path to the correct tsconfig in your package.json).

https://github.com/mikro-orm/mikro-orm/blob/master/packages/core/src/utils/ConfigurationLoader.ts#L143-L148

**Comment by B4nan** [comment]
That project setup is quite weird, not sure if nest is handling the paths mapping somehow automatically, but I don't see any place where you would configure it. 

I'd rather suggest using relative paths, they work universally, don't depend on a 3rd party package, and will generally save you a lot of troubles.

I will move this into discussions, as it is clearly a configuration issue on your side. Will try to think about how to improve this in v6.
